src/pydantic_zarr/v3.py

```diff
@@ -229,25 +318,28 @@ def from_zarr(cls, array: zarr.Array) -> Self:
 
         """
         from zarr.core.metadata import ArrayV3Metadata
+        from zarr.core.metadata.v3 import V3JsonEncoder
 
         if not isinstance(array.metadata, ArrayV3Metadata):
             raise ValueError("Only zarr v3 arrays are supported")  # noqa: TRY004
-
-        print(array.compressors)
-
+        meta_json = json.loads(
+            json.dumps(array.metadata.to_dict(), cls=V3JsonEncoder), object_hook=tuplify_json
+        )
```

Comment:
this is very annoying -- although the `ArrayV3Metadata` object defines a `to_dict` method, it doesn't generate a dict with JSON-serializable values. So to get the JSON friendly version we have to literally write the metadata to JSON using a custom JSON encoder class. Definitely room for improvement on the zarr python side here.